src/main/java/gift/controller/product/OptionController.java

```diff
@@ -0,0 +1,65 @@
+package gift.controller.product;
+
+import gift.controller.product.dto.OptionRequest;
+import gift.controller.product.dto.OptionResponse;
+import gift.service.product.OptionService;
+import gift.service.product.dto.OptionModel;
+import java.util.List;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class OptionController {
+
+    private final OptionService optionService;
+
+    public OptionController(OptionService optionService) {
+        this.optionService = optionService;
+    }
+
+    @GetMapping("/products/{id}/options")
+    public ResponseEntity<OptionResponse.InfoList> getOptions(
+        @PathVariable("id") Long productId
+    ) {
+        List<OptionModel.Info> models = optionService.getOptions(productId);
+        OptionResponse.InfoList response = OptionResponse.InfoList.from(models);
+        return ResponseEntity.ok(response);
+    }
+
+    @PostMapping("/products/{id}/options")
+    public ResponseEntity<OptionResponse.InfoList> createOption(
+        @PathVariable("id") Long productId,
+        @RequestBody OptionRequest.Register request
```

Comment:
유효성 검사가 작동되도록 `@Valid`를 붙여주어야 할 것 같습니다ㅎㅎ
```suggestion
        @RequestBody @Valid OptionRequest.Register request
```

Reply:
중요한 부분을 놓쳤네요 감사합니다!